packages/mui-material/src/SvgIcon/SvgIcon.js

```diff
@@ -40,23 +40,34 @@ const SvgIconRoot = styled('svg', {
   display: 'inline-block',
   fill: 'currentColor',
   flexShrink: 0,
-  transition: theme.transitions.create('fill', {
-    duration: theme.transitions.duration.shorter,
-  }),
-  fontSize: {
-    inherit: 'inherit',
-    small: theme.typography.pxToRem(20),
-    medium: theme.typography.pxToRem(24),
-    large: theme.typography.pxToRem(35),
-  }[ownerState.fontSize],
+  transition:
+    theme.$$designSystem === 'joy'
```

Comment:
`$$` is borrowed from react `$$typeof` to indicate that this field is for internal usage.

I tried with `Object.defineProperty` but it does not work with nested theme, so I decided to go with simplest option.

Reply:
Why don't you use theme overrides?

Reply:
theme overrides does not work if I want to do this:
```js
// This is what I expect to use in Joy
<Icon fontSize="sm" />
```